### ISSUE TITLE
Add 'browser' module resolution, supporting native JS modules

### ISSUE THREAD

**Issue by RafaelSalguero** [microsoft/TypeScript#37971]
## Search Terms

Disable non-relative modules
Disable absolute module path
Change node_modules folder
Disable node_modules non-relative module lookup

## Suggestion

A compiler option to disable non-relative modules, since these are invalid in the browser, 

May be

- An option to disable `node_modules` lookup
- Or a new `moduleResolution` option: `browser`, with the same behavior as `node` but without looking into `node_modules`

## Use Cases

Now that native JS modules are becoming popular is possible to create web apps without any bundling and/our module path rewriting. Currently this is possible but typescript allows non-relative modules, leaving the task of checking all imports for any invalid path to the user.

[Here](https://github.com/microsoft/TypeScript/issues/15479#issuecomment-300240856) is stated that per design one should write the import path that works at runtime, so is sensible to ask for a way to check for path validation considering that the code will run as-is on the browser

## Examples

```ts
//No problem! Native JS modules supported via pathMapping:
import * from "/web_modules/redux.js"

//OK too
import * from "/src/ui/button.js"; 

//WRONG! Non-relative modules are not supported in browser, but typescript allows it (should raise compiler error)
import * from "react";
```

## Checklist

My suggestion meets these guidelines:

* [ X ] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [ X ] This wouldn't change the runtime behavior of existing JavaScript code
* [ X ] This could be implemented without emitting different JS based on the types of the expressions
* [ X ] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, etc.)
* [ X ] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

**Comment by corbane** [comment]
hello,
whether for browsers or not, `baseUrl` should be able to specify the root" / "

so with `"baseUrl": "./out"`  in `./tsconfog.json`,
`import "/js/core.js";` in `./packages/app/src/index.ts` should map `../../../out/js/core.js`

`baseUrl` specifies the directory of non-relative paths. In fact I do not understand why in a project it would be possible to refer to the root of the computer with "/"
first, because everywhere it is recommended to include dependencies in the "node_modules" of the project.
and for a node project, the resolution of the modules is hierarchical (parent to parent)

The name of `baseUrl` is strange. This is not a URL but a computer path.
At first I thought it was a dedicated browser option like "<base>" or "document.baseURI" in a webpage.

conclusion:
`import "js/core.js";` or
`import "/js/core.js";` should work the same

jmv.

**Comment by ultimaweapon** [comment]
Any updates on this?
